source/js/update-recipe.js

```diff
@@ -1,20 +1,421 @@
+  const STEP_REQUIRED = "Please enter your recipe instructions";
+
+  // Form submission, saves recipe to localStorage
+  form.addEventListener("submit", function (event) {
+    // Stop form submission
+    event.preventDefault();
+    console.log("Submit button clicked");
+    let allValid = true;
+
+    // Validate form
+    const titleValid = hasValue(
+      document.getElementById("recipeTitle"),
+      TITLE_REQUIRED
+    );
+    if (titleValid) {
+      recipe["title"] = document.getElementById("recipeTitle").value.trim();
+    } else {
+      allValid = false;
+    }
+
+    recipe["description"] = document
+      .getElementById("recipeDescription")
+      .value.trim();
+
+    const totalCost = document.getElementById("recipeCost").value.trim();
+    let cost = null;
+    if (totalCost.length > 0) {
+      if (hasFloat(document.getElementById("recipeCost"))) {
+        cost = parseFloat(totalCost);
+      } else {
+        allValid = false;
+      }
+    }
+    recipe["totalCost"] = cost;
+
+    recipe["img-url"] = document.querySelector("#img-spot > img").src;
+
+    const ingredients = [];
+    const ingredientElems = document
+      .getElementById("ingredients")
+      .getElementsByClassName("ingredient");
+    for (let i = 0; i < ingredientElems.length; i++) {
+      const ingElem = ingredientElems[i];
+      const recipeIng = {};
+      const ingredientNameValid = hasValue(
+        ingElem.querySelector(".ingredient-name > input"),
+        INGREDIENT_NAME_REQUIRED
+      );
+      const ingredientAmountValid = hasValue(
+        ingElem.querySelector(".ingredient-amount > input"),
+        INGREDIENT_AMOUNT_REQUIRED
+      );
+      recipeIng["name"] = ingElem
+        .querySelector(".ingredient-name > input")
+        .value.trim();
+      recipeIng["amount"] = ingElem
+        .querySelector(".ingredient-amount > input")
+        .value.trim();
+
+      if (ingredientNameValid && ingredientAmountValid) {
+        ingredients.push(recipeIng);
+      } else {
+        allValid = false;
+      }
+    }
+
+    recipe["ingredients"] = ingredients;
+
+    const steps = [];
+    const stepElems = document
+      .getElementById("steps")
+      .getElementsByClassName("step-sec");
+    for (let i = 0; i < stepElems.length; i++) {
+      const stepElem = stepElems[i];
+      console.log(`stepElem: ${stepElem}`);
+      const stepValid = hasValue(
+        stepElem.querySelector(".form-control"),
+        STEP_REQUIRED
+      );
+      if (stepValid) {
+        const step = stepElem.querySelector(".form-control").value.trim();
+        steps.push(step);
+      } else {
+        allValid = false;
+      }
+    }
+    recipe["steps"] = steps;
+
+    if (allValid) {
+      recipes[recipeID] = recipe;
+      localStorage.setItem("recipes", JSON.stringify(recipes));
+      window.alert("successfully updated a recipe!");
+      let currUrl = location.toString().replace("update-recipe.html?id=" + recipeID, "cook-book.html");
```

Comment:
Codacy has a fix for the issue: ['currUrl' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/cse110-fa21-group7/cse110-fa21-group7/pullRequest?prid=8446960)
```suggestion
      const currUrl = location.toString().replace("update-recipe.html?id=" + recipeID, "cook-book.html");
```